naga/src/back/hlsl/writer.rs

```diff
@@ -2207,6 +2207,35 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
 
                 writeln!(self.out, ");")?;
             }
+            crate::Statement::ImageAtomic {
+                image,
+                coordinate,
+                array_index,
+                fun,
+                value,
+            } => {
+                write!(self.out, "{level}")?;
+
+                let fun_str = fun.to_hlsl_suffix();
+                write!(self.out, "Interlocked{fun_str}(")?;
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/interlockedmax

Reply:
I think this may need to be suffixed by 64 like in https://github.com/JMS55/wgpu/blob/a391e14e5114c3ada42924408c60f725b6d9bcdc/naga/src/back/hlsl/writer.rs#L2020